File: task_03/python/prime.py
n = int(input("Enter a positive number: "))

if n < 0:
    print("Not valid")
else:
    primes = []
    for i in range(2, n + 1):
        is_prime = True
        for j in range(2, i):
            if i % j == 0:
                is_prime = False
                break
        if is_prime:
            primes.append(i)

    print("Prime numbers up to", n, "are:", primes)

File: task_03/haskell/prime.hs
import Control.Monad (forM_, when)

nestedForLoops :: Int -> IO ()
nestedForLoops rows = do
  forM_ [2 .. rows] $ \i -> do
    let isprime = all (\j -> i `mod` j /= 0) [2 .. i-1]
    when isprime $ putStrLn $ show i

main :: IO ()
main = do
  putStrLn "Enter an integer: "
  input <- getLine
  let number = read input :: Int
  if number <= 0
    then putStrLn "Not valid"
    else nestedForLoops number
  
